packages/@ember/-internals/owner/index.ts

```diff
@@ -99,5 +101,16 @@ export function getOwner(object: any): Owner {
   @public
 */
 export function setOwner(object: any, owner: Owner): void {
+  if (DEBUG) {
+    let publicApiType = owner as any;
+
+    assert(`Expected owner to be truthy`, publicApiType);
```

Comment:
this doesn't include the entirety of the `Owner`, but what I left out is optional anyway (`?:`)